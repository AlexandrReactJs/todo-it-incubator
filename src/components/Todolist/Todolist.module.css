.Todolist{
    background-color: #fff;
    border-radius: 10px;
    padding: 16px;
    display: flex;
    flex-direction: column;
    width: 18%;
    margin-right: 20px;
}



.taskTitle{
    margin-bottom: 14px;
}

/*-----------------------------------------------------------*/

.newTask{
    margin-bottom: 14px;
}


.newTaskMaker{
    display: flex;
    flex-direction: row;
    position: relative;
}

.inputTitle{
    outline: none;
    border: unset;
    border-bottom: 2px solid rgb(255, 228, 193);
    height: 30px;
    width: 100%;
    font-size: 16px;
    padding: 0 4px;
}

.newTaskMaker img{
    width: 20px;
    height: 20px;
    position: absolute;
    top: 20%;
    left:92%;
    transition: all 0.2s linear;
}


.newTaskMaker img:hover{
    opacity: 0.3;
}


.error{
    border: unset;
    outline: none;
    height: 30px;
    border-bottom: 2px solid red
}

.errorMessage{
    color: red;
}



/*------------------------------------------------------------*/

.tasks{
    margin-bottom: 14px;
}

.tasksList{
    list-style: none;
    margin-left: 10px;
    min-height: 170px;
}

.task{
    display: flex;
    flex-direction: row;
  
    padding: 5px 0;
}

.taskCheckbox{
  margin-right: 10px;
}

.taskTitle{
    margin-right: 5px;
   
}

.taskTitle span{
    font-size: 16px;
}


.taskCancel img{
    width: 18px;
    height: 18px;
}

.taskCancel:hover{
    cursor: pointer;
}

.isDone{
    opacity: 0.5;
}

/*------------------------------------------*/







.filterButton{
    padding: 8px 14px;
    margin-right: 10px;
    background-color: rgb(214, 175, 124);
    border: unset;
    border-radius: 20px;
    color: #fff;
    font-weight: 600;
    font-size: 16px;
    transition: all 0.2s linear;
}

.filterButton:hover{
    padding: 8px 14px;
    margin-right: 10px;
    background-color: rgb(255, 255, 255);
    border: unset;
    border-radius: 20px;
    color: black;
    font-weight: 600;
    font-size: 16px;
    border: 1px solid black;
}

.activeFilter{
    padding: 8px 14px;
    margin-right: 10px;
    background-color: rgb(214, 175, 124);
    border: unset;
    border-radius: 20px;
    color: #fff;
    font-weight: 600;
    font-size: 16px;
    border: 1px solid black;
}






/* Rectangle 8 */

