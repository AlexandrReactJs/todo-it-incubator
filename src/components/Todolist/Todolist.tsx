import React, { ChangeEvent, KeyboardEvent } from "react";
import styles from './Todolist.module.css'
import addIcon from '../../assets/icons/add.png'
import cancelIcon from '../../assets/icons/cancel.png'

export type TasksType = {
    id: string
    title: string
    isDone: boolean
}

type PropsType = {
    title: string
    tasks: Array<TasksType>
    removeTask: Function
    setFilter: Function
    addTask: Function
    changeStatus: (taskId: string, isDone: boolean) => void
    filter: string
}




const Todolist = (props: PropsType) => {



    const [newTaskTitle, setNewTaskTitle] = React.useState<string>("")
    const [error, setError] = React.useState<boolean>(false)

    const addTaks = () => {
        if (newTaskTitle?.trim() === "") {
            setError(true)
        } else {
            props.addTask(newTaskTitle)
            setNewTaskTitle("")

        }

    }

    const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
        setNewTaskTitle(e.target.value)
        setError(false)
    }

    const onKeyDownHandler = (e: KeyboardEvent<HTMLInputElement>) => {
        if (e.key === "Enter") {
            addTaks()
        }
    }

    const onAllClickHandler = () => props.setFilter("all")
    const onActiveClickHandler = () => props.setFilter("active")
    const onCompletedClickHandler = () => props.setFilter("completed")


    if(!props.tasks) {
        return(
            <div className={styles.Todolist}>
            <div className={styles.taskTitle}>
                <h3>{props.title}</h3>
            </div>

            <div className={styles.newTask}>
                <div className={styles.newTaskMaker}>
                    <input value={newTaskTitle}
                        onChange={onChangeHandler}
                        type="text"
                        onKeyDown={onKeyDownHandler}
                        className={error ? styles.error : styles.inputTitle}
                        placeholder="Enter title"
                    />
                    <img onClick={addTaks} src={addIcon} alt="add task" />
                </div>

                {
                    error && <div>
                        <p className={styles.errorMessage}>Title is require</p>
                    </div>
                }

            </div>
            <div className={styles.tasks}>
                <p>No TASK</p>
            </div>
            <div>
                <button className={props.filter === "all" ? styles.activeFilter : styles.filterButton} onClick={onAllClickHandler}>All</button>
                <button className={props.filter === "active" ? styles.activeFilter : styles.filterButton} onClick={onActiveClickHandler}>Active</button>
                <button className={props.filter === "completed" ? styles.activeFilter : styles.filterButton} onClick={onCompletedClickHandler}>Completed</button>
            </div>



        </div>
        )
    }



    
    return (
        <div className={styles.Todolist}>
            <div className={styles.taskTitle}>
                <h3>{props.title}</h3>
            </div>

            <div className={styles.newTask}>
                <div className={styles.newTaskMaker}>
                    <input value={newTaskTitle}
                        onChange={onChangeHandler}
                        type="text"
                        onKeyDown={onKeyDownHandler}
                        className={error ? styles.error : styles.inputTitle}
                        placeholder="Enter title"
                    />
                    <img onClick={addTaks} src={addIcon} alt="add task" />
                </div>

                {
                    error && <div>
                        <p className={styles.errorMessage}>Title is require</p>
                    </div>
                }

            </div>
            <div className={styles.tasks}>
                <ul className={styles.tasksList}>
                    {
                        
                            props.tasks.map(el => {

                                const removeTaskHandler = () => {
                                    props.removeTask(el.id)
                                }

                                const changeStatus = (e: ChangeEvent<HTMLInputElement>) => {
                                    props.changeStatus(el.id, e.currentTarget.checked)
                                }

                                return <div className={el.isDone ? styles.isDone : ""}>
                                    <li className={styles.task} key={el.id}>
                                        <div className={styles.taskCheckbox}>
                                            <input type="checkbox" onChange={changeStatus} checked={el.isDone} />
                                        </div>
                                        <div className={styles.taskTitle}>
                                            <span>{el.title}</span>
                                        </div>
                                        <div className={styles.taskCancel}>
                                            <img onClick={removeTaskHandler} src={cancelIcon} alt="cancel task" />
                                        </div>
                                    </li>
                                </div>

                            }
                            )


                    }

                </ul>
            </div>
            <div>
                <button className={props.filter === "all" ? styles.activeFilter : styles.filterButton} onClick={onAllClickHandler}>All</button>
                <button className={props.filter === "active" ? styles.activeFilter : styles.filterButton} onClick={onActiveClickHandler}>Active</button>
                <button className={props.filter === "completed" ? styles.activeFilter : styles.filterButton} onClick={onCompletedClickHandler}>Completed</button>
            </div>



        </div>
    )
}




export default Todolist;