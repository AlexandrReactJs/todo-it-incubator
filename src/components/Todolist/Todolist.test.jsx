import { screen, render } from "@testing-library/react";
import { v1 } from 'uuid';
import Todolist from "./Todolist";

let tasks1 = [
    { id: v1(), title: 'React', isDone: true },
    { id: v1(), title: 'JS', isDone: true },
    { id: v1(), title: 'CSS', isDone: false }

  ]

describe("Todolist component", () => {
    it("Todolist renders", () => {
        render(<Todolist tasks={tasks1}/>)
    
        expect(screen.getByText("React")).toBeInTheDocument()
    })

    it("Todolist renders withoutdata", () => {
        render(<Todolist/>)

        expect(screen.queryByRole('list')).toBeNull()
    })

        it("TodoLis snapshot", () => {
            const todo = render(<Todolist tasks={tasks1}/> )
            expect(todo).toMatchSnapshot()
        })
})
