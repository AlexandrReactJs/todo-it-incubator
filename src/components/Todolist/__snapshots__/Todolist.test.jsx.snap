// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todolist component TodoLis snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Todolist"
      >
        <div
          class="taskTitle"
        >
          <h3 />
        </div>
        <div
          class="newTask"
        >
          <div
            class="newTaskMaker"
          >
            <input
              class="inputTitle"
              placeholder="Enter title"
              type="text"
              value=""
            />
            <img
              alt="add task"
              src="add.png"
            />
          </div>
        </div>
        <div
          class="tasks"
        >
          <ul
            class="tasksList"
          >
            <div
              class="isDone"
            >
              <li
                class="task"
              >
                <div
                  class="taskCheckbox"
                >
                  <input
                    checked=""
                    type="checkbox"
                  />
                </div>
                <div
                  class="taskTitle"
                >
                  <span>
                    React
                  </span>
                </div>
                <div
                  class="taskCancel"
                >
                  <img
                    alt="cancel task"
                    src="cancel.png"
                  />
                </div>
              </li>
            </div>
            <div
              class="isDone"
            >
              <li
                class="task"
              >
                <div
                  class="taskCheckbox"
                >
                  <input
                    checked=""
                    type="checkbox"
                  />
                </div>
                <div
                  class="taskTitle"
                >
                  <span>
                    JS
                  </span>
                </div>
                <div
                  class="taskCancel"
                >
                  <img
                    alt="cancel task"
                    src="cancel.png"
                  />
                </div>
              </li>
            </div>
            <div
              class=""
            >
              <li
                class="task"
              >
                <div
                  class="taskCheckbox"
                >
                  <input
                    type="checkbox"
                  />
                </div>
                <div
                  class="taskTitle"
                >
                  <span>
                    CSS
                  </span>
                </div>
                <div
                  class="taskCancel"
                >
                  <img
                    alt="cancel task"
                    src="cancel.png"
                  />
                </div>
              </li>
            </div>
          </ul>
        </div>
        <div>
          <button
            class="filterButton"
          >
            All
          </button>
          <button
            class="filterButton"
          >
            Active
          </button>
          <button
            class="filterButton"
          >
            Completed
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Todolist"
    >
      <div
        class="taskTitle"
      >
        <h3 />
      </div>
      <div
        class="newTask"
      >
        <div
          class="newTaskMaker"
        >
          <input
            class="inputTitle"
            placeholder="Enter title"
            type="text"
            value=""
          />
          <img
            alt="add task"
            src="add.png"
          />
        </div>
      </div>
      <div
        class="tasks"
      >
        <ul
          class="tasksList"
        >
          <div
            class="isDone"
          >
            <li
              class="task"
            >
              <div
                class="taskCheckbox"
              >
                <input
                  checked=""
                  type="checkbox"
                />
              </div>
              <div
                class="taskTitle"
              >
                <span>
                  React
                </span>
              </div>
              <div
                class="taskCancel"
              >
                <img
                  alt="cancel task"
                  src="cancel.png"
                />
              </div>
            </li>
          </div>
          <div
            class="isDone"
          >
            <li
              class="task"
            >
              <div
                class="taskCheckbox"
              >
                <input
                  checked=""
                  type="checkbox"
                />
              </div>
              <div
                class="taskTitle"
              >
                <span>
                  JS
                </span>
              </div>
              <div
                class="taskCancel"
              >
                <img
                  alt="cancel task"
                  src="cancel.png"
                />
              </div>
            </li>
          </div>
          <div
            class=""
          >
            <li
              class="task"
            >
              <div
                class="taskCheckbox"
              >
                <input
                  type="checkbox"
                />
              </div>
              <div
                class="taskTitle"
              >
                <span>
                  CSS
                </span>
              </div>
              <div
                class="taskCancel"
              >
                <img
                  alt="cancel task"
                  src="cancel.png"
                />
              </div>
            </li>
          </div>
        </ul>
      </div>
      <div>
        <button
          class="filterButton"
        >
          All
        </button>
        <button
          class="filterButton"
        >
          Active
        </button>
        <button
          class="filterButton"
        >
          Completed
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
